contrib/kafka/pkg/reconciler/resources/receive_adapter.go

```diff
@@ -73,11 +74,38 @@ func MakeReceiveAdapter(args *ReceiveAdapterArgs) *v1.Deployment {
 
 	env = appendEnvFromSecretKeyRef(env, "KAFKA_NET_SASL_USER", args.Source.Spec.Net.SASL.User.SecretKeyRef)
 	env = appendEnvFromSecretKeyRef(env, "KAFKA_NET_SASL_PASSWORD", args.Source.Spec.Net.SASL.Password.SecretKeyRef)
-
 	env = appendEnvFromSecretKeyRef(env, "KAFKA_NET_TLS_CERT", args.Source.Spec.Net.TLS.Cert.SecretKeyRef)
 	env = appendEnvFromSecretKeyRef(env, "KAFKA_NET_TLS_KEY", args.Source.Spec.Net.TLS.Key.SecretKeyRef)
 	env = appendEnvFromSecretKeyRef(env, "KAFKA_NET_TLS_CA_CERT", args.Source.Spec.Net.TLS.CACert.SecretKeyRef)
 
+	RequestResourceCPU, err := resource.ParseQuantity(args.Source.Spec.Resources.Requests.ResourceCPU)
+	if err != nil {
+		RequestResourceCPU = resource.MustParse("250m")
```

Comment:
Should this be an error instead of silently ignoring the value and choosing a default? For example, if my spec has `5000n` (when I intended `5000m`), would I expect the KafkaSource to have a failing status or to create a receive adapter with 250m?

Reply:
@Harwayne I tried implementing this (as per our discussion yesterday on slack), but I'm finding that when I specify an invalid resource amount (even after implementing a corresponding `KafkaStatus*` status functions) it seems that the controller fails to even run to the point where the Status functions would be triggered.  My best guess is that perhaps the admission webhook? might be already verifying the validity of the deployment's `corev1.ResourceRequirements` ?

```
ERROR: logging before flag.Parse: E0524 15:06:51.020134       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.KafkaSource: v1alpha1.KafkaSourceList.Items: []v1alpha1.KafkaSource: v1alpha1.KafkaSource.Spec: v1alpha1.KafkaSourceSpec.Sink: Resources: v1.ResourceRequirements.Requests: unmarshalerDecoder: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$', error found in #10 byte of ...|pu":"500p"}},"sink":|..., bigger context ...|},"key":{}}},"resources":{"requests":{"cpu":"500p"}},"sink":{"apiVersion":"serving.knative.dev/v1alp|...
```
Is there another approach you'd like to explore?  Or, with the change to `corev1.ResourceRequirements` can we consider this point resolved?

Reply:
Was that failing CO created before or after the change to `corev1.ResourceRequirements`? If it was before, then it might be something that the webhook/API server would correctly handle for us now. If it was after, then that is a big problem and we might need to revert to your original implementation.

Reply:
CO?  I submitted that (erroneous) `resource.yaml` file (which lead to that error message) after the change to `corev1.ResourceRequirements`

Reply:
CO = Custom Object. An actual instance of a CRD. For example, Broker = CRD, the Broker named 'default' in the namespace 'default' = CO.

Was that log from the webhook or the controller?

Reply:
Gotcha, thanks!  the output was produced via controller logs;
`k -n knative-sources logs kafka-controller-manager-0`

Reply:
I don't know why it is failing in this way, but this not a failure we can tolerate, so we may need to revert to your earlier implementation of hand rolling the equivalent of `corev1.ResourceRequirements`. 

@n3wscott @vaikas-google any ideas why this is failing in this way? Namely that the API server seems to accept something that can't be unmarshaled into the Go object.

Reply:
you are setting `quantities` to `"500p"` and the regex is `'^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'`

try changing it to `"500P"` ?

Reply:
Thanks @n3wscott while that works, I was under the impression that wasn't what Adam was asking about.  Perhaps I've misunderstood @Harwayne 's initial review comment.  Are we trying to handle;
1. The case where the resourcespec itself is incorrect (ie, 500p -- lowercase p) OR
2. The case where the underlying resourcespec matches the correct syntax, but is not appropriate for the platform its running on? (ie, 500G on my laptop)?

I thought it was the former?

Reply:
This initial comment thread was about what happens when the value is invalid. The initial code silently replaced it with a default value. I argued it should set a status error message instead.

Another thread recommended replacing the hand rolled resource requirements with the actual `corev1.ResourceRequirements`, but that seemed to cause weird failures _only_ in the controller, which really scares me. So if we can't find the cause of that (such that bad values are rejected either before getting to the controller, or the controller handles them gracefully), then I said we should move back to the hand rolled resource requirements.

Reply:
@Harwayne thanks for the clarification.  Unless @n3wscott (or anybody else) suggests otherwise, I'm happy to start reverting to my previous implementation.

Reply:
Now that we have reverted, I think my original question still stands:

> Should this be an error instead of silently ignoring the value and choosing a default? For example, if my spec has 5000n (when I intended 5000m), would I expect the KafkaSource to have a failing status or to create a receive adapter with 250m?

Essentially, can this function return `(*v1.Deployment, error)` rather than silently ignoring bad values?

Reply:
I understand now. Before this is called, `checkResourcesStatus(src)` is called which verifies the fields. 

Perhaps change these to panics, rather than defaulting? We expect them to never happen, and I would rather they fail loudly if they do. However, I'll leave that decision to you.